File: packages/python/src/alumnium/accessibility/accessibility_element.py
from dataclasses import dataclass
from typing import Any


@dataclass
class AccessibilityElement:
    id: int | None = None
    backend_node_id: int | None = None
    name: str | None = None
    label: str | None = None
    type: str | None = None
    value: str | None = None
    androidresourceid: str | None = None
    androidclass: str | None = None
    androidtext: str | None = None
    androidcontentdesc: str | None = None
    androidbounds: str | None = None
    frame: Any | None = None  # Playwright Frame object for iframe support
    frame_chain: list[int] | None = None  # For Selenium: chain of iframe backendNodeIds from root to element's frame

File: packages/typescript/src/accessibility/AccessibilityElement.ts
export interface AccessibilityElement {
  id?: number | undefined;
  backendNodeId?: number | undefined;
  name?: string | undefined;
  label?: string | undefined;
  type?: string | undefined;
  value?: string | undefined;
  androidResourceId?: string | undefined;
  androidClass?: string | undefined;
  androidText?: string | undefined;
  androidContentDesc?: string | undefined;
  androidBounds?: string | undefined;
  maestroBounds?: string | undefined;
  frame?: object | undefined;
  frameChain?: number[] | undefined; // For Selenium: chain of iframe backendNodeIds from root to element's frame
}
